maven-cli/src/main/java/org/apache/maven/cling/invoker/LookupInvoker.java

```diff
@@ -413,6 +413,7 @@ protected void helpOrVersionAndMayExit(C context) throws Exception {
             } else {
                 context.terminal.writer().println(CLIReportingUtils.showVersion());
             }
+            context.terminal.writer().flush();
```

Comment:
I would add a flush after displaying the help, just so that we don't rely on `displayHelp` to call `flush`...